### ISSUE TITLE
Scaled yolo 4 large p5  values always NAN in training although  ADAM optimised is set 0.001 and 0.0001 

### ISSUE THREAD

**Issue by xhumanz** [WongKinYiu/ScaledYOLOv4#247]
**Command** 
!python train.py --batch-size 8 --img 896 --epoch 50 --data '../data.yaml' --cfg './models/yolov4-p5.yaml' --weights ''  --name 'yolop4-p5' --adam --hyp './data/hyp.scratch.yaml'

**values always NAN in training 
although all images annotations well** 

am using Colab GPU
Using CUDA device0 _CudaDeviceProperties(name='Tesla P100-PCIE-16GB', total_memory=16280MB)

Namespace(adam=True, batch_size=8, bucket='', cache_images=False, cfg='/content/gdrive/MyDrive/CustomYOLO/ScaledYOLOv4/models/yolov4-p5.yaml', data='/content/gdrive/MyDrive/CustomYOLO/data.yaml', device='', epochs=50, evolve=False, global_rank=-1, hyp='/content/gdrive/MyDrive/CustomYOLO/ScaledYOLOv4/data/hyp.scratch.yaml', img_size=[896, 896], local_rank=-1, logdir='runs/', multi_scale=False, name='hmd_p5', noautoanchor=False, nosave=False, notest=False, rect=False, resume=False, single_cls=False, sync_bn=False, total_batch_size=8, weights='', world_size=1)
Start Tensorboard with "tensorboard --logdir runs/", view at http://localhost:6006/
2021-05-22 20:12:18.609887: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
Hyperparameters {'lr0': 0.001, 'momentum': 0.937, 'weight_decay': 0.0005, 'giou': 0.05, 'cls': 0.5, 'cls_pw': 1.0, 'obj': 1.0, 'obj_pw': 1.0, 'iou_t': 0.2, 'anchor_t': 4.0, 'fl_gamma': 0.0, 'hsv_h': 0.015, 'hsv_s': 0.7, 'hsv_v': 0.4, 'degrees': 0.0, 'translate': 0.5, 'scale': 0.5, 'shear': 0.0, 'perspective': 0.0, 'flipud': 0.0, 'fliplr': 0.5, 'mixup': 0.0}
Overriding /content/gdrive/MyDrive/CustomYOLO/ScaledYOLOv4/models/yolov4-p5.yaml nc=80 with nc=1     

![image](https://user-images.githubusercontent.com/84167046/119242550-07ad0980-bb5f-11eb-979f-494630a92de9.png)

**Comment by xhumanz** [comment]
@WongKinYiu

**Comment by WongKinYiu** [comment]
does sgd works well?

**Comment by xhumanz** [comment]
the same it gives me NAN
!python train.py --batch-size 16 --img 896 --epoch 50 --data '../data.yaml' --cfg './models/yolov4-p5.yaml' --weights ''  --name 'yolop5' --hyp './data/hyp.scratch.yaml' 

lr0: 0.01  # initial learning rate (SGD=1E-2, Adam=1E-3)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
giou: 0.05  # GIoU loss gain
cls: 0.5  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 1.0  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 0.0  # image rotation (+/- deg)
translate: 0.5  # image translation (+/- fraction)
scale: 0.5  # image scale (+/- gain)
shear: 0.0  # image shear (+/- deg)
perspective: 0.0  # image perspective (+/- fraction), range 0-0.001
flipud: 0.0  # image flip up-down (probability)
fliplr: 0.5  # image flip left-right (probability)
mixup: 0.0  # image mixup (probability)
![image](https://user-images.githubusercontent.com/84167046/119251411-ad379b80-bba6-11eb-9247-e28422adc03f.png)
![image](https://user-images.githubusercontent.com/84167046/119251480-0e5f6f00-bba7-11eb-8146-740b1d9c3073.png)
@WongKinYiu

**Comment by xhumanz** [comment]
@WongKinYiu

**Comment by jorgegaticav** [comment]
@xhumanz did you manage to make it work ?

**Comment by shrysh8bit** [comment]
What is the solution to the NaN issue? Can't find any solution but the issue has been closed.
